chips/nrf52/src/qdec.rs

```diff
@@ -0,0 +1,315 @@
+    /// of the interrupt register bits are set. If it
+    /// is, then put it in the client's bitmask
+    /// TODO: DO I NEED TO DISABLE INTERRUPTS
+    /// TODO: ADD CLIENT CODE TO HIL/CAPSULE/LIB
+    pub fn handle_interrupt(&self) {
+        self.disable_interrupts();
+        let regs = &*self.registers;
+        self.client.map(|client| {
+            let mut val = 0;
+            // For each of 4 possible compare events, if it's happened,
+            // clear it and sort its bit in val to pass in callback
+            for i in 0..4 { // TODO: either add events_compare or add each individual register
+                if self.registers.events_arr[i].is_set(Event::READY) {
+                    val = val | 1 << i;
+                    self.registers.events_arr[i].write(Event::READY::CLEAR);
+                    // Disable corresponding interrupt
+                    let interrupt_bit = match i {
+                        0 => Inte::SAMPLERDY::SET,
+                        //1 => Inte::REPORTRDY::SET,
+                        //2 => Inte::ACCOF::SET,
+                        //3 => Inte::DBLRDY::SET,
+                        4 => Inte::STOPPED::SET,
+                        _ => panic!("Unknown interrupt value!"),
+                    };
+                    if i == 0 {
+                        client.sample_ready (); 
+                    } else if i == 4 && self.qdec.state == SAMPLE_RATE_STOP {
+                        debug!("Received stopped signal!");
+                        regs.sample_per.write(SampPer::SAMPLEPER.val(5));
+                        regs.task_start.write(Task::ENABLE::SET);
+                    }
+                }
+            }
+        });
+        self.enable_interrupts();
+    }
+
+    // NOTE: ALL INTERRUPTS ARE ONLY FOR SAMPLING RIGHT NOW 
+    fn enable_interrupts(&self) {
+        let regs = &*self.registers;
+        regs.intenset.write(Inte::SAMPLERDY::SET);
+        regs.intenset.write(Inte::STOPPED::SET); /*SET SAMPLE READY*/
+    }
+
+    fn disable_interrupts(&self) {
+        let regs = &*self.registers;
+        regs.intenclr.write(Inte::SAMPLERDY::SET);
+    }
+
+    fn set_sample_rate (&self) { 
+        regs.task_stop.write(Task::ENABLE::SET);
+        self.qdec.state = SAMPLE_RATE_STOP;
+        regs.tasks_stop.write(Task::ENABLE::SET); /*induce stop*/
+    }
+
+    fn enable(&self) {
+        let regs = &*self.registers;
+        regs.enable.write(Task::ENABLE::SET);
+        set_sample_rate(); /*set sample rate to maximum sample rate*/
+        regs.tasks_start.write(Task::ENABLE::SET);
+        debug!("Enabled!");
+    }
+
+    fn is_enabled(&self) -> ReturnCode {
+        let regs = &*self.registers;
+        let result = if regs.enable.is_set(Task::ENABLE) {
+                        ReturnCode::SUCCESS
+                     } else {
+                        ReturnCode::FAIL
+                     };
+        result
+    }
+
+}    
+
+//TODO: FIX SPACING!
```

Comment:
running `make formatall` from the root directory should fix all of the spacing issues